gateway/mw_streaming.go

```diff
@@ -0,0 +1,383 @@
+			path, ok := httpPathMap[pathItem]
+			if !ok {
+				continue
+			}
+
+			if pathString, ok := path.(string); ok {
+				paths = append(paths, pathString)
+			}
+		}
+	}
+
+	return paths
+}
+
+func (s *StreamingMiddleware) getStreamsConfig(r *http.Request) *StreamsConfig {
+	if !s.Spec.IsOAS {
+		return nil
+	}
+
+	tykStreamingConfig, ok := s.Spec.OAS.T.Extensions[ExtensionTykStreaming]
+	if !ok {
+		return nil
+	}
+
+	data, err := json.Marshal(tykStreamingConfig)
+	if err != nil {
+		return nil
+	}
+
+	streamsConfig := &StreamsConfig{}
+	err = json.Unmarshal(data, streamsConfig)
+	if err != nil {
+		return nil
+	}
+
+	for streamID, stream := range streamsConfig.Streams {
+		if r != nil {
+			s.Logger().Debugf("Stream config for %s: %v", streamID, stream)
+
+			marshaledStream, err := json.Marshal(stream)
+			if err != nil {
+				s.Logger().Errorf("Failed to marshal stream config: %v", err)
+				continue
+			}
+			replacedStream := s.Gw.replaceTykVariables(r, string(marshaledStream), true)
+
+			if replacedStream != string(marshaledStream) {
+				s.Logger().Debugf("Stream config changed for %s: %s", streamID, replacedStream)
+			} else {
+				s.Logger().Debugf("Stream config has not changed for %s: %s", streamID, replacedStream)
+			}
+
+			var unmarshaledStream map[string]any
+			err = json.Unmarshal([]byte(replacedStream), &unmarshaledStream)
+			if err != nil {
+				s.Logger().Errorf("Failed to unmarshal replaced stream config: %v", err)
+				continue
+			}
+			stream = unmarshaledStream
+		} else {
+			s.Logger().Debugf("No request available to replace variables in stream config for %s", streamID)
+		}
+	}
+	return streamsConfig
+}
+
+// createStream creates a new stream
+func (sm *StreamManager) createStream(streamID string, config map[string]interface{}) error {
+	streamFullID := fmt.Sprintf("%s_%s", sm.mw.Spec.APIID, streamID)
+	sm.mw.Logger().Debugf("Creating stream: %s", streamFullID)
+
+	stream := streaming.NewStream(sm.mw.allowedUnsafe)
+	err := stream.Start(config, &handleFuncAdapter{
+		mw:       sm.mw,
+		streamID: streamFullID,
+		muxer:    sm.muxer,
+		sm:       sm,
+		// child logger is necessary to prevent race condition
+		logger: sm.mw.Logger().WithField("stream", streamFullID),
+	})
+	if err != nil {
+		sm.mw.Logger().Errorf("Failed to start stream %s: %v", streamFullID, err)
+		return err
+	}
+
+	sm.streams.Store(streamFullID, stream)
+	sm.mw.Logger().Infof("Successfully created stream: %s", streamFullID)
+
+	return nil
+}
+
+func (sm *StreamManager) hasPath(path string) bool {
+	for _, p := range sm.listenPaths {
+		if strings.TrimPrefix(path, "/") == strings.TrimPrefix(p, "/") {
+			return true
+		}
+	}
+	return false
+}
+
+// ProcessRequest will handle the streaming functionality
+func (s *StreamingMiddleware) ProcessRequest(w http.ResponseWriter, r *http.Request, _ interface{}) (error, int) {
+	strippedPath := s.Spec.StripListenPath(r.URL.Path)
+	if !s.defaultStreamManager.hasPath(strippedPath) {
```

Comment:
If I'm not missing something, we only use `defaultStreamManager` to call `hasPath` method. Decoupling config management and the business logic could be a good idea to improve code readability and maintainability.

Reply:
yes i agree, we need to decouple that logic, but we are still not sure of the full scope and use of the defaultManager for portal so i postponed it to a future refactor PR